### PR TITLE
auth functions to use fetch

### PR DESCRIPTION
**Pull request by aaronli03**
## What
- xhr requests for auth related functions now use fetch
- tests updated
- mocks updated

## Why
- fetch is a more modern, simpler, and cleaner approach to making requests
- tests reflect changes

## Impact
- Will customer experience be significantly altered? No
- Does the code change cover more than one task? No

## Testing
- tested locally by running `npm run dev`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
 	},
 	"type": "module",
 	"dependencies": {
+		"cross-fetch": "^4.0.0",
```
Comment (byjackli) [inline package.json:47]: please explain your research and reasoning behind why this is required - correct me if the linked webpage below is insufficient/incorrect 

https://developer.mozilla.org/en-US/docs/Web/API/fetch

File: src/api/spotify/auth.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import pkg from 'crypto-js';
 const { SHA256, enc } = pkg;
-import type { MyResponseType, LocalStorage } from './types';
+import type { LocalStorage } from './types';
+import fetch from 'cross-fetch';
```
Comment (byjackli) [inline src/api/spotify/auth.ts:4]: fetch seems like it is a web-included standard and this seems really sus

*ref comment above

File: package.json (hunk 1 above)
Comment (aaronli03) [reply package.json:47]: It is needed to run tests because fetch is not native to the Node.js enviornent.

File: src/api/spotify/auth.ts (hunk 2 above)
Comment (aaronli03) [reply src/api/spotify/auth.ts:4]: It is needed to run tests because fetch is not native to the Node.js enviornent.

**Review by byjackli — APPROVED** [review]

**Pull request merged**